civis/io/_tables.py

```diff
@@ -956,31 +1042,19 @@ def _get_sql_select(table, columns=None):
 
 
 def _get_headers(client, sql, database, credential_id, polling_interval=None):
-    # use 'begin read only;' to ensure we can't change state
```

Comment:
The latter; specifically, the assumption was made that Redshift would return results as an escaped-delimited file rather than a true CSV. This was, as I understand, only true when a a specific Platform feature flag was active. For more info please see the [JIRA for this change](https://civisanalytics.atlassian.net/browse/CIVP-19104).

Reply:
Two additional points here and for full disclosure I wrote the original code so I was shaking my head at myself:

1. The existing code is written based on implementation details in the Console Ruby code to attempt to optimize the performance. That is bad. In the future we should consider removing this optimization all together.

2. There was a lot of unnecessary complexity injected because of the use of the `include_header` parameter even though a user cannot pass that parameter to `civis_file_to_table`. IMHO the new implementation is much more succinct.

Reply:
That's helpful, thanks.